src/loops.rs

```diff
@@ -43,6 +47,24 @@ impl LintPass for LoopsPass {
                     }
                 }
             }
+
+            // check for looping over x.iter() or x.iter_mut(), could use &x or &mut x
+            if let ExprMethodCall(ref method, _, ref args) = arg.node {
+                // just the receiver, no arguments to iter() or iter_mut()
+                if args.len() == 1 {
+                    let method_name = method.node.name.as_str();
+                    let object = snippet(cx, args[0].span, "_");
```

Comment:
nit: we should only call `snippet()` when we're ready to print, or at least guarded by some conditions. This will call it for every method call in a for loop, which is inefficient (`snippet` does a file read and stuff IIRC)